app/src/main/java/com/arygm/quickfix/model/profile/Profile.kt

```diff
@@ -3,18 +3,43 @@ package com.arygm.quickfix.model.profile
 import com.google.firebase.Timestamp
 import com.google.firebase.firestore.GeoPoint
 
-data class Profile(
+open class Profile(
     val uid: String,
     val firstName: String,
     val lastName: String,
     val email: String,
     val birthDate: Timestamp,
     val description: String,
+    val location: GeoPoint? = null
+)
+
+class UserProfile(
     val isWorker: Boolean = false,
+    uid: String,
+    firstName: String,
+    lastName: String,
+    email: String,
+    birthDate: Timestamp,
+    description: String,
+    location: GeoPoint? = null
+) : Profile(uid, firstName, lastName, email, birthDate, description, location)
+
+class WorkerProfile(
     val fieldOfWork: String? = null,
     val hourlyRate: Double? = null,
-    val location: GeoPoint? = null
-)
+    uid: String,
+    firstName: String,
+    lastName: String,
```

Comment:
I believe placing `firstName` and `lastName` in the `WorkerProfile` is a good idea because it simplifies the `WorkerProfileViewModel` by allowing it to have only one repository field, rather than two. Others might argue that since a worker is inherently a user, the `firstName` and `lastName` should be stored directly in the `UserProfile` of the worker. However, I think both approaches are equivalent in practice.